backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+	fullURL  string // full path URL
+	target   string // symlink target when filetype is symlink
+}
+
+//------------------------------------------------------------------------------
+
+// Stat is an object which holds the information of the stat element of the response xml
+type Stat struct {
+	XMLName   xml.Name `xml:"stat"`
+	Files     []File   `xml:"file"`
+	Directory string   `xml:"directory,attr"`
+}
+
+// File is an object which holds the information of the file element of the response xml
+type File struct {
+	XMLName    xml.Name `xml:"file"`
+	Type       string   `xml:"type,attr"`
+	Name       string   `xml:"name,attr"`
+	NameBase64 string   `xml:"name_base64,attr"`
+	Size       string   `xml:"size,attr"`
+	Md5        string   `xml:"md5,attr"`
+	Mtime      string   `xml:"mtime,attr"`
+	Bytes      string   `xml:"bytes,attr"`
+	Files      string   `xml:"files,attr"`
+	Target     string   `xml:"target,attr"`
+}
+
+// List is an object which holds the information of the list element of the response xml
+type List struct {
+	XMLName xml.Name   `xml:"list"`
+	Files   []File     `xml:"file"`
+	Resume  ListResume `xml:"resume"`
+}
+
+// ListResume represents the resume xml element of the list
+type ListResume struct {
+	XMLName xml.Name `xml:"resume"`
+	Start   string   `xml:"start,attr"`
+}
+
+// Du represents the du xml element of the response
+type Du struct {
+	XMLName   xml.Name `xml:"du"`
+	Directory string   `xml:"directory,attr"`
+	Duinfo    DuInfo   `xml:"du-info"`
+}
+
+// DuInfo represents the du-info xml element of the response
+type DuInfo struct {
+	XMLName xml.Name `xml:"du-info"`
+	Files   string   `xml:"files,attr"`
+	Bytes   string   `xml:"bytes,attr"`
+}
+
+// ResponseObject defines an interface for the response which can return the xml element name
+type ResponseObject interface {
+	GetName() xml.Name
+}
+
+// GetName returns a normalized name of the Stat item
+func (s Stat) GetName() xml.Name {
+	return s.XMLName
+}
+
+// GetName returns a normalized name of the List item
+func (l List) GetName() xml.Name {
+	return l.XMLName
+}
+
+// GetName returns a normalized name of the Du item
+func (d Du) GetName() xml.Name {
+	return d.XMLName
+}
+
+//------------------------------------------------------------------------------
+
+// NewFs creates a new Fs object from the name and root. It connects to
+// the host specified in the config file.
+//
+// If root refers to an existing object, then it should return an Fs which
+// points to the parent of that object and ErrorIsFile.
+func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, error) {
+	// Parse config into Options struct
+	opt := new(Options)
+	err := configstruct.Set(m, opt)
+	if err != nil {
+		fs.Errorf(nil, "NewFs name: %s root: %s couldn't parse config into struct: %v", name, root, err)
+		return nil, err
+	}
+
+	//the base URL (endPoint is protocol + :// + domain/internal folder
+	opt.Endpoint = opt.Protocol + "://" + opt.Endpoint
```

Comment:
Normally we just put URLs in the config without separating the protocol. That would probably look neater here just to have `endpoint = https://......`